### PR TITLE
Huge clothing thickness audit and guidelines

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
None

#### Purpose of change
Completion of #46932 
I messed up and failed to set the previous PR to draft when I realized I needed to add some guidelines, and then as I was adding guidelines I found some errors I had made, and then as I was fixing those I realized I really just needed to audit all the clothes.

#### Describe the solution
Did some research and formed some guidelines for thicknesses.
Audited all the clothing items against these guidelines.

#### Describe alternatives you've considered
I've pretty much left all the "serious" armor alone, this is aimed at a problem I see in that all the "ordinary clothes" have inflated armor values because the dynamic range for thickness has been so narrow.

#### Testing
Handfulls of unit tests around some emblematic "ordinary clothes" having extremely small armor values.
Ideally we'd have some reasonably extensive spot checking of specific pieces of armor.  If you have an argument for some particular pieces having particular values, please make it and I can see about incorporating some spot checking.

#### Additional context
I'm going to leave this in draft for a bit to elicit some feedback.
Mostly what I'm looking for are either factual "well this piece is actually so-and-so thick" or "here's how much protection this should offer and here's why".

How about some data?  Here are the thicknesses of various items that have changed dramatically:

< 0.1mm | 0.1mm | 0.2mm | 0.3mm - 0.4mm | 0.5mm - 0.9mm | 1.0mm | 2.0mm
--------|-------|-------|---------------|---------------|-------|------
maid_hat | sheet | drinking_hat | thermal_socks | holo_cloak | harmonica_holder | xedra_seismograph |
plastic_shopping_bag | blindfold | eboshi | thermal_gloves | fedora | blanket | acoustic_guitar |
debug_backpack | folding_poncho_on | hat_newsboy | thermal_mask | turtleneck | down_blanket | guitar_electric |
veil_wedding | emer_blanket_on | hat_sombrero | towel | turtleneck_rolled | quilt | bagpipes |
socks_bag | rad_monitor | porkpie | towel_wet | turtleneck_shirt | arm_splint | wallet_travel |
gloves_bag | thermal_outfit | straw_hat | patchwork_scarf | turtleneck_shirt_rolled | tourniquet_upper | fur_blanket |
boxer_shorts | bandana_head | tophat | patchwork_scarf_loose | hat_knit | ear_plugs | welding_mask_crude |
under_armor | chaplet | tricorne | long_patchwork_scarf | dump_pouch | leg_splint | welding_mask_crude_raised |
unitard | hat_ball | legrig | long_patchwork_scarf_loose | cloak_wool | thermal_suit | binoculars |
leotard | hat_chef | makeshift_knapsack | 10gal_hat | optical_cloak | wearable_light | rm13_armor |
bikini_bottom | hat_cotton | ragpouch | beret | helmet_liner | survivor_light | dimensional_anchor |
| hat_noise_cancelling | rucksack | beret_wool | slippers | wearable_atomic_light | phase_immersion_suit |
| kippah | clown_wig | bowhat | chainmail_harness_dog | rebreather | mask_filter |
| kufi | socks | cowboy_hat | loincloth | rebreather_xl | mask_gas |
| postman_hat | tabi_dress | hat_boonie | nanoskirt | straw_fedora | mask_gas_xl |
| hat_golf | tabi_gi | hat_fur | winter_pants_army | knit_scarf | goggles_nv |
| fanny | hot_pants_fur | hat_hunting | zubon_gi | knit_scarf_loose | goggles_ir |
| leather_pouch | jeans | turban | balclava | long_knit_scarf | mask_h20survivor |
| runner_bag | jeans_red | american_flag | bondage_mask | scarf | mask_h20survivorxl |
| tarp | pants | cape_fp | mask_rioter | scarf_loose | long_knit_scarf_loose |
| stockings | pants_army | jedi_cloak | gloves_fingerless | scarf_long | scarf_fur |
| stockings_tent_legs | pants_cargo | socks_ankle | gloves_fingerless_mod | scarf_long_loose | scarf_fur_loose |
| rubber_harness_dog | pants_checkered | socks_wool | glove_jackson | mask_ski | scarf_fur_long |
| headscarf | postman_shorts | antarvasa | gloves_leather | mask_ski_loose | scarf_fur_long_loose |
| b_shorts | shorts | kilt | gloves_golf | solarpack_on | powered_earmuffs |
| hot_pants | shorts_cargo | technician_pants_gray | arm_warmers | electric_blanket | powered_earmuffs_on |
| hot_pants_leather | shorts_denim | gown | bikini_top_fur | foodperson_mask | ammo_satchel |
| leggings | striped_pants | jersey | bikini_top_leather | ammo_pouch | chestpouch |
| skirt | mask_guy_fawkes | duster | chestwrap_leather | quiver | chestrig |
| bandana | jumpsuit | hakama_gi | shirt_straw | quiver_birchbark | legpouch |
| mask_dust | subsuit_xl | hakama | jacket_light | quiver_large | legpouch_large |
| clown_suit | union_suit | jacket_army | judo_gi | quiver_large_birchbark | tacvest |
| zentai | gloves_light | jacket_windbreaker | karate_gi | bindle | backpack |
| gloves_liner | long_glove_white | keikogi | kariginu | daypack | backpack_hiking |
| gloves_medical | sockmitts | haori | kasaya | dive_bag | backpack_giant |
| army_top | stockings_tent_arms | kittel | samghati | makeshift_sling | backpack_tactical_large |
| bikini_top | chestwrap | sleeveless_duster | sleeveless_duster_fur | mbag | basket_laundry |
| boxer_briefs | chestwrap_fur | tunic | tux | net_backpack | bigback |
| boy_shorts | chestwrap_wool | tunic_rag | waistcoat | purse | briefcase |
| bra | leg_warmers_f | hood_rain | cloak_fur | slingpack | duffelbag |
| briefs | long_underpants | vest | pants_leather | swag_bag | jerrypack |
| camisole | long_undertop | cloak | skirt_leather | camera_bag | molle_pack |
| leg_warmers | long_undertop_sleeveless | breeches | | mouthpiece | petpack |
| panties | dress | uttarasanga | | lame_foil | straw_basket |
| thermal_shirt | dress_wedding | jacket_chef | | lame_saber | survivor_duffel_bag |
| sports_bra | flag_shirt | | | axe_ring | survivor_pack |
| tank_top | longshirt | | | baldric | survivor_rucksack |
| tights | polo_shirt | | | bootsheath | survivor_runner_pack |
| under_armor_shorts | sheriffshirt | | | bootsheath_birchbark | travelpack |
| undershirt | striped_shirt | | | bscabbard | wicker_backpack |
| dress_shirt | technician_shirt_gray | | | scabbard | long_duffelbag |
| halter_top | coat_lab | | | sheath | rifle_case_soft |
| linuxtshirt | coat_rain | | | sheath_birchbark | rifle_case_soft_leather |
| maid_dress | coat_gut | | | spearsling | rifle_case_soft_2 |
| postman_shirt | jacket_flannel | | | cloak_leather | rifle_case_soft_leather_2 |
| sundress | jacket_jean | | | poncho | rifle_case_xl_soft_leather |
| tshirt | kimono | | | ear_spool | crown_golden |
| tshirt_text | | | | garnet_platinum_tiara | crown_golden_survivor |
| jacket_evac | | | | garnet_gold_tiara | cuirass_scrap |
| yukata | | | | garnet_silver_tiara | snuggie |
| hood_gut | | | | dance_shoes | pot_helmet |
| trunks | | | | flip_flops | helmet_scrap |
| | | | | flip_flops_exp | pot_xlhelmet |
| | | | | footrags | bastsandals |
| | | | | footrags_fur | boots |
| | | | | footrags_leather | boots_combat |
| | | | | footrags_wool | boots_hiking |
| | | | | heels | boots_scrap |
| | | | | leathersandals | boots_steel |
| | | | | lowtops | cleats |
| | | | | mocassins | golf_shoes |
| | | | | shoes_birchbark | clownshoes |
| | | | | sneakers | dress_shoes |
| | | | | straw_sandals | geta |
| | | | | nomex_socks | knee_high_boots |
| | | | | eclipse_glasses | motorbike_boots |
| | | | | eyepatch_leather | shoes_bowling |
| | | | | fancy_sunglasses | swim_fins |
| | | | | fitover_sunglasses | thigh_high_boots |
| | | | | glasses_bifocal | wetsuit_booties |
| | | | | glasses_eye | roller_shoes_off |
| | | | | glasses_monocle | roller_shoes_on |
| | | | | glasses_reading | leg_warmers_xlf |
| | | | | iggaak | kevlar_harness |
| | | | | sunglasses | glasses_safety |
| | | | | sunglasses_bifocal | goggles_ski |
| | | | | sunglasses_eye | fencing_pants |
| | | | | sunglasses_reading | legguard_scrap |
| | | | | fancy_glasses_bifocal | armor_scrapsuit |
| | | | | fancy_glasses_eye | beekeeping_suit |
| | | | | fancy_glasses_reading | hazmat_suit |
| | | | | fancy_sunglasses_bifocal | nomex_suit |
| | | | | fancy_sunglasses_eye | robofac_enviro_suit |
| | | | | fancy_sunglasses_reading | suit |
| | | | | transition_glasses_bifocal | fishing_waders |
| | | | | transition_glasses_eye | mask_hockey |
| | | | | transition_glasses_reading | survivor_belt_notools |
| | | | | fancy_transition_glasses_bifocal | tool_belt |
| | | | | fancy_transition_glasses_eye | beekeeping_gloves |
| | | | | fancy_transition_glasses_reading | chainmail_hands |
| | | | | cowl_wool | gauntlet_fencing |
| | | | | keffiyeh | gauntlet_fencing_l |
| | | | | bunker_pants | gloves_survivor_fingerless |
| | | | | chaps_cut_resistant | gloves_work |
| | | | | motorbike_pants | gloves_cut_resistant |
| | | | | cleansuit | gloves_wraps |
| | | | | stillsuit | gloves_wraps_fur |
| | | | | kilt_leather | gloves_wraps_leather |
| | | | | loincloth_fur | gloves_wraps_wool |
| | | | | loincloth_leather | gloves_xlsurvivor |
| | | | | loincloth_wool | winter_gloves_army |
| | | | | pants_fur | gloves_studded |
| | | | | pants_ski | survivor_vest |
| | | | | skirt_grass | dragonskinempty |
| | | | | fencing_mask | leg_warmers_xl |
| | | | | leather_belt | armguard_scrap |
| | | | | bondage_suit | armguard_soft |
| | | | | jumpsuit_xl | apron_leather |
| | | | | wool_suit | flotation_vest_ms |
| | | | | gloves_lsurvivor | coat_fur |
| | | | | gloves_tactical | coat_fur_sf |
| | | | | mittens | coat_straw |
| | | | | nomex_gloves | coat_winter |
| | | | | back_holster | house_coat |
| | | | | baldric_holster | robe |
| | | | | bootstrap | ski_jacket |
| | | | | bow_sling | sleeveless_duster_survivor |
| | | | | holster | sleeveless_trenchcoat |
| | | | | XL_holster | thawb |
| | | | | blazer | trenchcoat |
| | | | | hoodie | saddlebag |
| | | | | sweater | grenadebandolier |
| | | | | sweatshirt | wetsuit_spring |
| | | | | vest_leather | wetsuit_gloves |
| | | | | wool_hoodie | survivor_belt |
| | | | | bunker_coat | |
| | | | | cassock | |
| | | | | duster_leather | |
| | | | | jacket_leather | |
| | | | | jacket_leather_red | |
| | | | | peacoat | |
| | | | | sleeveless_duster_leather | |
| | | | | sleeveless_trenchcoat_leather | |
| | | | | sleeveless_tunic | |
| | | | | trenchcoat_leather | |
| | | | | winter_jacket_army | |
| | | | | beekeeping_hood | |
| | | | | bandolier_pistol | |
| | | | | bandolier_rifle | |
| | | | | bandolier_shotgun | |
| | | | | torso_bandolier_shotgun | |
| | | | | flintlock_pouch | |
| | | | | bandolier_wrist | |
| | | | | grenade_pouch | |
| | | | | stone_pouch | |
| | | | | gloves_h20survivor | |
| | | | | goggles_swim | |
| | | | | q_solarpack_on | |
| | | | | cestus | |

Side note, I got these item lists with a series of jq invocations like this
find data/json/ -name "*.json" -print0 | xargs -0 jq '.[]|select(has("id"))|select(.type|contains("ARMOR"))|select(has("material_thickness"))|select(.material_thickness >= 2.0)|select(.material_thickness < 3.0).id' > over_two

### CODE REVIEW

**Comment by UmbralReaper** [comment]
What's your source for tshirt thickness being <0.1mm. I'd put it at 0.1 at least, and a [random google](https://propercloth.com/reference/fabric-thickness-weight/) seems to agree.

**Comment by actual-nh** [comment]
Regarding a cestus: This depends a lot on what you're talking about, and they look to frequently have been variable thickness (more on some parts than others) - the maximum thickness with solid evidence I can (quickly) find is [this statue](https://en.wikipedia.org/wiki/Boxer_at_Rest), which is actually thicker than the fingers, but covers very little with that thickness - further (both hands and partial arm coverings) would be thinner.

**Comment by kevingranade** [comment]
There's an error in deriving my chart, the tshirts are all placed at 0.1mm, which is supported by 
https://kalamazoospiritshoppe.com/fast-guide-common-t-shirt-fabric-differences/ that puts a "standard" tshirt weight at 4.5 - 5.3 oz, which puts them at the bottom of the range for cloth from the other source.  In other words, yes I agree they should be around 0.1mm, and they are, sorry for the noise.

Regarding the cestus,  yes I'm evaluating it as an armor, so the pertinent thichkness is whatever part has the most coverage.  If you want to go back and make it (and arguably all of the hand wraps?) one thickness for the forearm part and another thickness for the hand part, that would be great, but I'm not doing that kind of fine-tuning here.

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -2388,6 +2388,31 @@ Alternately, every item (book, tool, gun, even food) can be used as armor if it
...
+Chart cobbled together from several sources
+Very lightweight fabrics are between 0-4oz. (up to 136 gsm) 0.006″-0.007″ 0.15mm
+Light fabrics are between 4-7oz. (up to 237 gsm) 0.008″ 0.2mm
+Medium fabrics are between 7-11oz. (up to 373 gsm) 0.009″-0.011" 0.25mm
+Heavy fabrics are between 11-14oz. (up to 475 gsm) 0.012″-0.014″ 0.3mm
+
+Shoe thicknesses outlined at https://secretcobbler.com/choosing-leather/
+TL;DR upper 1.2 - 2.0mm, lining 0.8 - 1.2mm, for a total of 2.0 - 3.2mm
+
+Reference for turnout gear http://bolivar.mo.us/media/uploads/2014/09/2014-06-bid-fire-gear-packet.pdf
```
Comment (actual-nh) [inline doc/JSON_INFO.md:2414]:
```suggestion
#### Guidelines for thickness: ####
According to [this site](https://propercloth.com/reference/fabric-thickness-weight/), dress shirts and similar fine clothing range from 0.15mm to 0.35mm.
According to [this site](https://leathersupreme.com/leather-hide-thickness-in-leather-jackets/):
* Fashion leather such as thin leather jackets, skirts, and thin vests are 1.0mm or less.
* Heavy leather clothes such as motorcycle suits average 1.5mm.
From [this site](http://cci.one/site/marine/design-tips-fabrication-overview/tables-of-weights-and-measures/?doing_wp_cron=1611725817.5850219726562500000000), an equivalency guideline for fabric weight to mm:

| Cloth | g/m2 | Inches | mm |
| ----- | ------ | ------ | ---- |
| 2.3 oz/yd2 fiberglass (plain weave cloth) | 78 | 0.004 | 0.10 |
| 6.0 oz/yd2 fiberglass (plain weave cloth) | 203 | 0.007 | 0.17 |
| 5.0 oz/yd2 Kevlar (TM) (plain weave cloth) | 170 | 0.010 | 0.25 |
| 5.8 oz/yd2 Carbon Fiber (plain weave cloth) | 197 | 0.009 | 0.23 |
| 9.0 oz/yd2 Carbon Fiber (unidirectional)  | 305 | 0.011 | 0.28 |

Chart cobbled together from several sources:
* Very lightweight fabrics are 0-4oz. (up to 136 gsm): 0.006″-0.007″ - use 0.15mm
* Light fabrics are 4-7oz. (up to 237 gsm): 0.008″ - use 0.2mm
* Medium fabrics are 7-11oz. (up to 373 gsm): 0.009″-0.011" - use 0.25mm
* Heavy fabrics are 11-14oz. (up to 475 gsm): 0.012″-0.014″ - use 0.3mm

Shoe thicknesses are outlined at [this site](https://secretcobbler.com/choosing-leather/); TL;DR: upper 1.2 - 2.0mm, lining 0.8 - 1.2mm, for a total of 2.0 - 3.2mm

For turnout gear, see [this reference](http://bolivar.mo.us/media/uploads/2014/09/2014-06-bid-fire-gear-packet.pdf).
```
This isn't quite finished - I should probably turn the last listing into a table also - but it's a start.

Comment (kevingranade) [reply doc/JSON_INFO.md:2414]: Thanks a ton! :1st_place_medal:

Comment (actual-nh) [reply doc/JSON_INFO.md]: Quite welcome; happy to help!

**Pull request merged**